### PR TITLE
NJ SNAP minimum allotment

### LINKED ISSUE #4405: New Jersey SNAP minimum benefit
$95/mo

https://www.nj.gov/humanservices/njsnap/apply/eligibility/

Per https://github.com/PolicyEngine/policyengine-us/issues/3610#issuecomment-1897481580

### PR DESCRIPTION
**Pull request by Kiki-67**
Fix #4405

### CODE REVIEW

File: policyengine_us/variables/gov/usda/snap/snap.py [hunk 1]
```diff
@@ -27,5 +27,6 @@ def formula(spm_unit, period, parameters):
                     "snap_normal_allotment",
                     "snap_emergency_allotment",
                     "dc_snap_temporary_local_benefit",
+                    "nj_snap_min_allotment",
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/snap.py:30]: I think we should add this to the `snap_min_allotment` with the NJ condition

File: policyengine_us/variables/gov/usda/snap/nj_snap_min_allotment.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    unit = USD
+    defined_for = StateCode.NJ
+
+    def formula(spm_unit, period, parameters):
+        p = parameters(period).gov.usda.snap.temporary_local_benefit.nj
+        base_benefit = spm_unit('snap_reported', period)
+        if base_benefit < p.amount:
+            return p.amount
+        else:
+            return 0
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/nj_snap_min_allotment.py:22]: Could we change this to a where statement

File: policyengine_us/variables/gov/usda/snap/nj_snap_min_allotment.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+from policyengine_us.model_api import *
+
+
+class nj_snap_min_allotment(Variable):
+    value_type = float
+    entity = SPMUnit
+    definition_period = MONTH
+    documentation = "NJ minimum snap payment"
+    label = "NJ minimum snap payment"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/nj_snap_min_allotment.py:9]:
```suggestion
    label = "New Jersey minimum snap payment"
```

File: policyengine_us/parameters/gov/usda/snap/temporary_local_benefit/nj/amount.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+description: NJ provides a minimum monthly SNAP benefit at this amount
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/usda/snap/temporary_local_benefit/nj/amount.yaml:1]:
```suggestion
description: New Jersey provides a minimum monthly SNAP benefit of this amount.
```

File: policyengine_us/parameters/gov/usda/snap/temporary_local_benefit/nj/amount.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/usda/snap/temporary_local_benefit/nj/amount.yaml:1]: Could you add a bool parameter for new jersey applies a separate min allotment - 2023: true - before false

File: policyengine_us/parameters/gov/usda/snap/temporary_local_benefit/nj/amount.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+values:
+  2023-01-01: 95
+metadata:
+  period: month
+  unit: currency-USD
+  label: NJ SNAP minimum allotment amount
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/usda/snap/temporary_local_benefit/nj/amount.yaml:7]:
```suggestion
  label: New Jersey SNAP minimum allotment amount
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: changelog_entry.yaml [hunk 6]
```diff
@@ -0,0 +1 @@
+
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: need changelog

File: policyengine_us/parameters/gov/usda/snap/temporary_local_benefit/nj/in_effect.yaml [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+values:
+  0000-01-01: false
+  2023-03-01: true
+metadata:
+  period: year
+  unit: bool
+  label: New Jersey SNAP minimum allotment in effect
+  reference:
+      - title: NJ SNAP
+      href: https://www.nj.gov/humanservices/njsnap/emergency/covid19/
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/usda/snap/temporary_local_benefit/nj/in_effect.yaml:11]:
```suggestion
    - title: New Jersey Department of Human Services - SNAP
      href: https://www.nj.gov/humanservices/njsnap/emergency/covid19/
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/usda/snap/temporary_local_benefit/nj/in_effect.yaml:11]: all tests

File: policyengine_us/variables/gov/usda/snap/snap_min_allotment.py [hunk 8]
```diff
@@ -15,13 +15,22 @@ def formula(spm_unit, period, parameters):
...
+
         # Minimum benefits only apply to households up to a certain size.
         size = spm_unit("spm_unit_size", period)
         eligible = size <= min_allotment.maximum_household_size
-        return eligible * min_allotment.rate * relevant_max_allotment
+        base_benefit = eligible * min_allotment.rate * relevant_max_allotment
+
+        # Check if the state is in New Jersey then apply specific minimum payment
+        state_code = spm_unit.household("state_code", period)
+        is_nj = state_code == StateCode.NJ
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/snap_min_allotment.py:33]:
```suggestion
        in_nj = state_code == StateCode.NJ
```

File: policyengine_us/variables/gov/usda/snap/snap_min_allotment.py [hunk 9]
```diff
@@ -15,13 +15,22 @@ def formula(spm_unit, period, parameters):
...
+        nj_supplement = spm_unit("nj_snap_min_allotment", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/snap_min_allotment.py:34]:
```suggestion
        nj_min_allotment = spm_unit("nj_snap_min_allotment", period)
```

File: policyengine_us/variables/gov/usda/snap/nj_snap_min_allotment.py [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+    definition_period = MONTH
+    documentation = "NJ minimum snap payment"
+    label = "New Jersey minimum snap payment"
+    reference = "https://www.nj.gov/humanservices/njsnap/apply/eligibility/"
+    unit = USD
+    defined_for = StateCode.NJ
+
+    def formula(spm_unit, period, parameters):
+        p = parameters(period).gov.usda.snap.temporary_local_benefit.nj
+        base_benefit = spm_unit("snap_reported", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/nj_snap_min_allotment.py:16]:
```suggestion
        snap_amount = spm_unit("snap_reported", period)
```

File: policyengine_us/variables/gov/usda/snap/snap_min_allotment.py [hunk 11]
```diff
@@ -15,13 +15,22 @@ def formula(spm_unit, period, parameters):
...
         ]
+
         # Minimum benefits only apply to households up to a certain size.
         size = spm_unit("spm_unit_size", period)
         eligible = size <= min_allotment.maximum_household_size
-        return eligible * min_allotment.rate * relevant_max_allotment
+        base_benefit = eligible * min_allotment.rate * relevant_max_allotment
+
+        # Check if the state is in New Jersey then apply specific minimum payment
+        state_code = spm_unit.household("state_code", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/snap_min_allotment.py:32]: need to use the in effect parameter

File: policyengine_us/variables/gov/usda/snap/snap_min_allotment.py [hunk 12]
```diff
@@ -15,13 +15,22 @@ def formula(spm_unit, period, parameters):
...
         snap_region = spm_unit.household("snap_region_str", period)
         relevant_max_allotment = snap.max_allotment.main[snap_region][
             str(min_allotment.relevant_max_allotment_household_size)
         ]
+
         # Minimum benefits only apply to households up to a certain size.
         size = spm_unit("spm_unit_size", period)
         eligible = size <= min_allotment.maximum_household_size
-        return eligible * min_allotment.rate * relevant_max_allotment
+        base_benefit = eligible * min_allotment.rate * relevant_max_allotment
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/snap_min_allotment.py:29]:
```suggestion
        base_benefit_amount = eligible * min_allotment.rate * relevant_max_allotment
```

File: policyengine_us/variables/gov/usda/snap/nj_snap_min_allotment.py [hunk 13]
```diff
@@ -0,0 +1,17 @@
+from policyengine_us.model_api import *
+
+
+class nj_snap_min_allotment(Variable):
+    value_type = float
+    entity = SPMUnit
+    definition_period = MONTH
+    documentation = "NJ minimum snap payment"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/nj_snap_min_allotment.py:8]:
```suggestion
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/usda/snap/snap_min_allotment.py [hunk 14]
```diff
@@ -15,13 +15,34 @@ def formula(spm_unit, period, parameters):
...
+            period
+        ).gov.usda.snap.temporary_local_benefit.nj.amount
+        in_nj = state_code == "NJ"
+        nj_min_allotment_eligible = in_nj & nj_min_allotment_in_effect
+
+        return where(
+            nj_min_allotment_eligible,
+            nj_min_allotment_amount,
+            base_benefit_amount,
+        )
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/snap_min_allotment.py:48]:
```suggestion
        if snap.temporary_local_benefit.nj.in_effect
             state_code = spm_unit.household("state_code_str", period)
             in_nj = state_code == "NJ"
             nj_min_allotment = snap.temporary_local_benefit.nj.amount
             base_benefit_amount = where(in_nj, nj_min_allotment, base_benefit_amount)
        return base_benefit_amount
```

File: policyengine_us/parameters/gov/usda/snap/temporary_local_benefit/nj/amount.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/usda/snap/temporary_local_benefit/nj/amount.yaml:1]: can we move this file to parameters/states/nj/snap/....

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/usda/snap/snap_min_allotment.py [hunk 15]
```diff
@@ -13,25 +13,27 @@ class snap_min_allotment(Variable):
...
         # Minimum benefits only apply to households up to a certain size.
         size = spm_unit("spm_unit_size", period)
-        eligible = size <= p.min_allotment.maximum_household_size
-        min_allotment = (
-            eligible * p.min_allotment.rate * relevant_max_allotment
+        eligible = size <= min_allotment.maximum_household_size
+        base_benefit_amount = (
+            eligible * min_allotment.rate * relevant_max_allotment
         )
-        p_dc = parameters(period).gov.states.dc.dhs.snap.min_allotment
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/snap_min_allotment.py:29]: we want to keep the DC changes and add NJ on top of it

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/tests/policy/baseline/gov/usda/snap/snap_min_allotment.yaml [hunk 16]
```diff
@@ -17,14 +17,23 @@
...
+  input:
+    spm_unit_size: 2
+    state_code: NJ
+    state_group: CONTIGUOUS_US
+  output:
+    snap_min_allotment: 95 * 12  
+
 - name: DC provides a different minimum allotment amount
-  period: 2024-01
+  period: 2024
```
Comment (MaxGhenis) [inline policyengine_us/tests/policy/baseline/gov/usda/snap/snap_min_allotment.yaml:30]: Can we keep the period as a month? For monthly variables it better aligns with our minimalist approach. Please change others in this file too.

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/tests/policy/baseline/gov/usda/snap/snap_min_allotment.yaml (hunk 16 above)
Comment (MaxGhenis) [reply policyengine_us/tests/policy/baseline/gov/usda/snap/snap_min_allotment.yaml]: >Please change others in this file too.

**Comment by MaxGhenis** [comment]
Pull again - too many changes

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**